src/Models/TeamGroups.php

```diff
@@ -65,42 +64,49 @@ public function getApiPath(): string
     #[Override]
     public function readAll(?QueryParamsInterface $queryParams = null): array
     {
-        $sql = "SELECT team_groups.id,
-                team_groups.name,
-                GROUP_CONCAT(users.userid ORDER BY users.firstname, users.lastname) AS userids,
-                GROUP_CONCAT(CONCAT(users.firstname, ' ', users.lastname) ORDER BY users.firstname, users.lastname SEPARATOR '|') AS fullnames
-            FROM team_groups
-            LEFT JOIN users2team_groups ON (
-                users2team_groups.groupid = team_groups.id
+        $sql = 'SELECT
+              tg.id,
+              tg.name,
+              u.userid,
+              u.firstname,
+              u.lastname
+
+            FROM team_groups AS tg
+            LEFT JOIN users2team_groups AS ug ON (
+                ug.groupid = tg.id
             )
-            LEFT JOIN users USING (userid)
-            WHERE team_groups.team = :team
-            GROUP BY team_groups.id
-            ORDER BY team_groups.name ASC";
+            LEFT JOIN users AS u ON ug.userid = u.userid
+            WHERE tg.team = :team
+            ORDER BY
+                tg.id,
+                CASE WHEN u.userid IS NULL THEN 1 ELSE 0 END,
+                u.firstname,
+                u.lastname';
 
         $req = $this->Db->prepare($sql);
         $req->bindParam(':team', $this->Users->userData['team'], PDO::PARAM_INT);
         $this->Db->execute($req);
 
-        $fullGroups = array();
+        $groups = array();
         while ($group = $req->fetch()) {
-            $fullGroups[] = array(
-                'id' => $group['id'],
-                'name' => $group['name'],
-                'users' => isset($group['userids'])
-                    ? array_map(
-                        fn(string $userid, ?string $fullname): array => array(
-                            'userid' => (int) $userid,
-                            'fullname' => $fullname,
-                        ),
-                        explode(',', $group['userids']),
-                        explode('|', $group['fullnames'])
-                    )
-                    : array(),
-            );
+            $gid = $group['id'];
+            if (!isset($groups[$gid])) {
+                $groups[$gid] = array(
+                    'id'    => $gid,
+                    'name'  => $group['name'],
+                    'users' => array(),
+                );
+            }
+            if ($group['userid'] !== null) {
+                $fullname = ($group['firstname'] ?? '') . ' ' . ($group['lastname'] ?? '');
```

Comment:
suggestion to prevent unwanted leading or trailing whitespace in the fullname string
```suggestion
                $fullname = trim(sprintf('%s %s', $group['firstname'] ?? '', $group['lastname'] ?? ''));
```
:warning: if using it, goes along with use sprintf; use trim;